src/util/request-builder.js

```diff
@@ -128,8 +128,10 @@ export default class RequestBuilder {
         }
       }
     };
-    request.open(this.method, this.url);
-    request.timeout = this.retryConfig.timeout || 0;
+    request.open(this.method, this.url, this.retryConfig.async);
+    if (this.retryConfig.async && this.retryConfig.timeout) {
```

Comment:
```suggestion
    if (this.retryConfig.async && typeof this.retryConfig.timeout === 'number') {
```
to catch `0`

Reply:
zero is default so not needed

Reply:
https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/timeout